### PR TITLE
Fix type annotation

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Pass `spmatrix` as a string, which is used only when type checking.

## Description of the changes
Currently, if you use `OptunaSearchCV` with default dependencies and `scikit-learn`, you will get following error:
```
NameError: name 'spmatrix' is not defined
```
This would be because `import pandas as pd` under `try_import()` context fails and `spmatrix` is left uimported. The error message will not let users know that they need to install `pandas`. With this patch, the error message is going to be as follows:
```
ModuleNotFoundError: No module named 'pandas'
```

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@eukaryo @gen740 Could you review this PR?

File: optuna/integration/sklearn.py [hunk 1]
```diff
@@ -48,10 +48,10 @@
...
     BaseEstimator = object  # NOQA
 
-ArrayLikeType = Union[List, np.ndarray, "pd.Series", spmatrix]
+ArrayLikeType = Union[List, np.ndarray, "pd.Series", "spmatrix"]
 OneDimArrayLikeType = Union[List[float], np.ndarray, "pd.Series"]
-TwoDimArrayLikeType = Union[List[List[float]], np.ndarray, "pd.DataFrame", spmatrix]
-IterableType = Union[List, "pd.DataFrame", np.ndarray, "pd.Series", spmatrix, None]
+TwoDimArrayLikeType = Union[List[List[float]], np.ndarray, "pd.DataFrame", "spmatrix"]
+IterableType = Union[List, "pd.DataFrame", np.ndarray, "pd.Series", "spmatrix", None]
 IndexableType = Union[Iterable, None]
```
Comment (gen740) [inline optuna/integration/sklearn.py:55]: This suggestion isn't directly related to this PR, but how about using `Optional`.
```suggestion
IndexableType = Optional[Iterable]
```

**Review by gen740 — APPROVED** [review]
I guess, these type aliases do not so much serve as helpful type annotations, rather they add unnecessary complexity to the code.

For example.
The `_num_samples` function takes `ArrayLikeType` but in l72 `TwoDimArrayLikeType` is passed, which is not compatible with `ArrayLikeType`.

For now, it might be best to proceed with the current changes and consider making corrections later on.

**Review by eukaryo — APPROVED** [review]
LGTM, and I agree with the comment above. Let me merge this PR.

**Pull request merged**